--- a/asiou/settings.py
+++ b/asiou/settings.py
@@ -57,6 +57,9 @@
 # User profile module
 AUTH_PROFILE_MODULE = 'asiou.accounts.models.ASIOU_U_UserProfile'

+# SQL Cache (Cacheops)
+from settings_cacheops import *
+
 # Local time zone for this installation. Choices can be found here:
 # http://en.wikipedia.org/wiki/List_of_tz_zones_by_name
 # although not all choices may be available on all operating systems.
@@ -226,6 +229,7 @@

 # List of installed apps
 INSTALLED_APPS = (
+    'cacheops',
     'piston',
     'django.contrib.auth',
     'django.contrib.contenttypes',
