--- a/asiou/export_rhd/utils.py
+++ b/asiou/export_rhd/utils.py
@@ -93,7 +93,7 @@
     # quality_calculate

     # Установим курсор
-    cursor = connection.cursor()
+    cursor = connections["for_reports"].cursor()
     dbname = connections.databases['default']['NAME']

     gia_participants = ASIOU_GIA_Participant.objects.filter()
@@ -181,8 +181,7 @@
             desc_name = dict(enumerate(desc_table))
             writer.writerow(desc_table)
             # Содержимое таблицы
-            # for row in cursor:
+            for row in cursor:
-            for row in cursor.fetchall():
                 if deface:
                     i = 0
                     param_flag = 0
@@ -227,21 +226,9 @@
         except Exception as e:
             if csvfile:
                 csvfile.close()
+    cursor.close()
     # Открываем BytesIO, чтобы захватить в памяти содержимое ZIP
     s = BytesIO()
-    # Объект архива
-    zf = zipfile.ZipFile(s, 'w', settings.APP_ZIP_COMPRESS)
-    for filename in filenames:
-        # Путь до файла во временной папке
-        fpath = os.path.join(settings.APP_RID_EXPORT_DIR, filename)
-        # Добавляем файл по заданному пути
-        zf.write(fpath, filename)
-    # Добавим в список файлов, файл с текущей версией БД
-    if type_rep == 'rep':
-        fpath = os.path.join(os.path.dirname(settings.APP_DBVER_FILE), 'options.ini')
-        zf.write(fpath, os.path.basename(fpath))
-    # Закрываем архив
-    zf.close()

     # Возьмем версию БД и дату сборки если есть
     # написал через рекурсию, можно через цикл, я думаю
@@ -260,4 +260,14 @@
     else:
         zipfile_name = 'exp-v{}-{}-{}.zip'.format(d_v, ou_id, datetime.now().strftime('%y%m%d-%H%M'))

+    # Для сжатия больших объёмов данных используем подходящий инструмент
+    from subprocess import check_call
+    full_filenames = [os.path.join(settings.APP_RID_EXPORT_DIR, fname) for fname in filenames]
+    full_filenames.append(settings.APP_DBVER_FILE)
+    if type_rep == 'rep':
+        full_filenames.append(os.path.join(os.path.dirname(settings.APP_DBVER_FILE), 'options.ini'))
+    zipfile_fullname = os.path.join(settings.APP_RID_EXPORT_DIR, zipfile_name)
+
+    check_call(["/usr/bin/zip", "-5", "-j", zipfile_fullname] + full_filenames)
+
     return zipfile_name, s
--- a/asiou/reports/views.py
+++ b/asiou/reports/views.py
@@ -1088,6 +1088,7 @@
     # Захватываем ZIP из памяти и отдаем его
     resp_dict = HttpResponse(s.getvalue(), content_type="application/x-zip-compressed")
     resp_dict['Content-Disposition'] = 'attachment; filename={}'.format(zipfile_name)
+    resp_dict['X-Accel-Redirect'] = '/temp-download/{}'.format(zipfile_name)

     return resp_dict

--- a/asiou/export_rhd/tasks.py
+++ b/asiou/export_rhd/tasks.py
@@ -8,6 +8,9 @@
 from asiou.export_rhd.utils import make_export_file
 from asiou.export_rhd.models import ASIOU_RHD_Export

+from django.conf import settings
+import os
+

 class PackageLoadCallback(task.Task):
     """ Класс позволяет выполнить какие либо действия по результатам выгрузки,
# @@ -53,12 +56,15 @@

     try:
         filename, result = make_export_file(None, export_type,institution.pk)
+        full_filename = os.path.join(settings.APP_RID_EXPORT_DIR, filename)
+        with open(full_filename, mode='rb') as file:
+            result = file.read()
     except:
         raise Exception

     # Постановка в очередь
     login, password, proto = _claim_get_user_passwd(institution)
-    files = {filename: result.getvalue()}
+    files = {filename: result}
     data = {'filename': filename}
     response = _do_rpc(
         'load_package', auth=(login, password), method='post', data=data,
